nexus_transporter/src/transporters/rmf_transporter.cpp

```diff
@@ -457,57 +622,62 @@ class RmfTransporter : public Transporter
         }
       });
 
-    _dispenser_request_sub = n->create_subscription<DispenserRequest>(
-      "/dispenser_requests",
-      100,
-      [&](DispenserRequest::SharedPtr msg)
-      {
-        auto n = _node.lock();
-        if (!n)
-        {
-          std::cerr << "RmfTransporter::_api_response_sub - invalid node"
-                    << std::endl;
-          return;
-        }
-        auto it = _rmf_task_id_to_ongoing_itinerary.find(msg->request_guid);
-        if (it == _rmf_task_id_to_ongoing_itinerary.end())
+    _dispenser_request_sub =
+      _internal_node->create_subscription<DispenserRequest>(
+        "/dispenser_requests",
+        100,
+        [&](DispenserRequest::SharedPtr msg)

```

Comment:
Cries a bit in merge conflicts for #101, I wonder if we make the diff a bit smaller by only moving the relevant task pub/subs into the internal node? I suppose that would be the api requests / responses?

Reply:
We can wait for #101 to be merged first, and I'll handle the merge conflicts later then. I think for now a quick look over the implementation is good enough.

I think it's better to isolate all the interactions to the internal node, so that we are sure that there will not be any race conditions or deadlocks if we use mutexes.